tests/function/test_config.py

```diff
@@ -190,3 +192,309 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who configs setting in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who and ignore the configurations files in
+            /etc/virt-who.d/ dir
+        """
+        config_file = '/tmp/test_config_configs.conf'
+        guest_uuid = hypervisor_handler.guest_uuid
+        globalconf.update('global', 'debug', 'True')
+        ssh_host.runcmd(f'\\cp -f {HYPERVISOR_FILE} {config_file}')
+
+        globalconf.update('global', 'configs', config_file)
+        result = virtwho.run_service()
+        msg = "ignoring configuration files in '/etc/virt-who.d/'"
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and guest_uuid in result['log']
+                and msg in result['log'])
+
+    def test_owner_in_virtwho_conf(self, virtwho, globalconf, hypervisor_handler, register_handler):
+        """Test the owner option in /etc/virtwho.conf
+
+        :title: virt-who: config: test owner option
+        :id: ce219d82-cf66-4019-af17-3197c53c72a0
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with owner setting in [defaults] section in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who with owner setting
+        """
+        guest_uuid = hypervisor_handler.guest_uuid
+        owner = register_handler.default_org
+        globalconf.update('global', 'debug', 'True')
+
+        globalconf.update('defaults', 'owner', owner)
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and guest_uuid in result['log'])
+
+    def test_hypervisor_id_in_virtwho_conf(self, virtwho, globalconf, hypervisor, hypervisor_handler):
+        """Test the hypervisor_id option in /etc/virtwho.conf
+
+        :title: virt-who: config: test hypervisor_id option
+        :id: fed463a6-9538-4242-9990-2e4995d1f473
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with hypervisor_id=hostname in /etc/virt-who.conf
+            2. Run virt-who with hypervisor_id=uuid in /etc/virt-who.conf
+            3. Run virt-who with hypervisor_id=hwuuid in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hostname
+            2. Succeeded to run the virt-who, the hypervisor_id in mapping info should be uuid
+            3. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hwuuid
+        """
+        host_uuid = hypervisor_handler.esx_uuid
```

Comment:
This test file will be run for the main hypervisor, but not only the `esx` mode, we can select which one or ones will be the main hypervisors.
So this is why I replace the `hypervisor_handler` by `hypervisor_data` fixture in conftest

Reply:
updated